internal/transactions/send.go

```diff
@@ -63,20 +64,52 @@ func send(
 ) (command.Result, error) {
 	codeFilename := args[0]
 
-	transactionSigner := sendFlags.Signer
-	if sendFlags.Signer == config.DefaultEmulatorServiceAccountName { // use service account by default
-		transactionSigner = state.Config().Emulators.Default().ServiceAccount
-	}
-	signer, err := state.Accounts().ByName(transactionSigner)
-	if err != nil {
-		return nil, err
+	var signed *flowkit.Transaction
+	var signers []*flowkit.Account
+
+	//validate all signers
+	for _, signerName := range sendFlags.Signer {
+		// use service account by default
+		if signerName == config.DefaultEmulatorServiceAccountName {
+			signerName = state.Config().Emulators.Default().ServiceAccount
+		}
+		signer, err := state.Accounts().ByName(signerName)
+		if err != nil {
+			return nil, fmt.Errorf("signer account: [%s] doesn't exists in configuration", signerName)
+		}
+		signers = append(signers, signer)
 	}
 
 	code, err := readerWriter.ReadFile(codeFilename)
 	if err != nil {
 		return nil, fmt.Errorf("error loading transaction file: %w", err)
 	}
 
+	//find authorizer count from code
+	authorizerCount := flowkit.GetAuthorizerCount(codeFilename, code)
+
+	var seen map[flow.Address]any = make(map[flow.Address]any)
+	var authorizers []flow.Address
+	for _, auth := range signers {
+		if _, ok := seen[auth.Address()]; ok {
+			continue
+		}
+		authorizers = append(authorizers, auth.Address())
+	}
+
+	if len(authorizers) < authorizerCount {
+		return nil, fmt.Errorf("invalid number of authorizers, expected: %d", authorizerCount)
+	}
+
+	//remove extra signers
+	authorizers = authorizers[:authorizerCount]
+
+	//proposer is the first signer
+	proposer := signers[0]
+
+	//payer signs last
+	payer := signers[len(signers)-1]
```

Comment:
I feel this should be encapsulated in the transaction model.

Reply:
in here https://github.com/onflow/flow-cli/blob/master/pkg/flowkit/transaction.go#L289

Reply:
@sideninja How about if we add something like `NewTransactionFromPayloadWithSigners(payload, signers)`?

I am worried about some confusion for flowkit users as a method. ( as it sets, payer, proposer and authorizers only from signers )

Reply:
Oh sorry I meant just the first part of validation of authorizer count, the second part of setting those values can stay there or be extracted in that method. I feel those methods will have to be refactored to build pattern soon probably anyway.